sql/api/src/main/scala/org/apache/spark/sql/util/ArrowUtils.scala

```diff
@@ -146,19 +146,33 @@ private[sql] object ArrowUtils {
           ArrowType.Struct.INSTANCE,
           null,
           Map("variant" -> "true").asJava)
+        val metadataFieldType = new FieldType(
```

Comment:
Do we have 2 ways of tagging something a variant? The line above is some metadata, but we also have this other metadata thing? This is confusing, so could you explain this more with some comments? Do we need both schemes?

Reply:
The line above should not have the variant tag. I have changed the design so that the metadata field has the variant tag. This is because Python codepaths don't have access to the top level field and metadata.

Thanks for catching.